src/static/js/modules/form-validation.js

```diff
@@ -6,34 +6,53 @@
 'use strict';
 
 var $ = require( 'jquery' );
+require( './jquery/cf_formValidator' ).init();
+require( './jquery/cf_notifier' ).init();
 
 function init() {
-  $( '.js-validate_form-not-empty' ).each( function() {
-    var $form = $( this );
-
-    $form.on( 'submit', function() {
-      var formIsEmpty = true;
-      var event;
-
-      $.each( $form.serializeArray(), function( index, element ) {
-        if ( element.value !== '' ) {
-          formIsEmpty = false;
-        }
-      } );
-
-      if ( formIsEmpty ) {
-        event = 'form:validate:empty';
-      } else {
-        event = 'form:validate:not_empty';
+  function sendNotification( elem, field ) {
+    var label = field.label;
+    var type;
+    var message;
+
+    for ( var prop in field.status ) {
+      if ( field.status[prop] === false ) {
+        type = prop;
       }
+    }
 
-      $form.trigger( event, $form );
+    if ( type === 'required' ) {
+      message = 'Please fill out the ' + label + ' field.';
+    }
 
-      return !formIsEmpty;
-    } );
+    if ( type === 'email' ) {
+      message = 'Please include a valid email address.';
+    }
+
+    if ( type === 'checkgroup' ) {
+      message = 'Please select at least one of the ' + label + ' options.';
+    }
 
-    return $form;
-  } );
+    $( elem ).trigger( 'cf_notifier:notify', {
+      message: message
+    } );
+    $( field.elem ).addClass( 'error' );
+  }
+
+  function clearNotification( elem ) {
+    $( elem ).trigger( 'cf_notifier:clear' );
+    $( elem ).find('.error').removeClass( 'error' );
+  }
+
+  $( 'form' ).cf_formValidator( 'init', {
+    onFailure: function( event, fields ) {
+      event.preventDefault();
+      sendNotification( this, fields.invalid[0] );
+    },
+    onSuccess: function( event, fields ) {
+      clearNotification( this );
+    }
+  } ).cf_notifier();
 }
 
 module.exports = { init: init };
```

Comment:
I'd suggest the following changes to modules:
- Don't nest function declarations when not necessary.
- Prefix with an underscore functions that will not be exported from the module.
- Make the `init` method only contain runtime initialization code (e.g. not function declarations).

I'd re-write this module to look like:

```js
'use strict';

var $ = require( 'jquery' );
require( './jquery/cf_formValidator' ).init();
require( './jquery/cf_notifier' ).init();

function init() {
  $( 'form' ).cf_formValidator( 'init', {
    onFailure: function( event, fields ) {
      event.preventDefault();
      _sendNotification( this, fields.invalid[0] );
    },
    onSuccess: function( event, fields ) {
      _clearNotification( this );
    }
  } ).cf_notifier();
}

function _sendNotification( elem, field ) {
  var label = field.label;
  var type;
  var message;

  for ( var prop in field.status ) {
    if ( field.status[prop] === false ) {
      type = prop;
    }
  }

  if ( type === 'required' ) {
    message = 'Please fill out the ' + label + ' field.';
  }

  if ( type === 'email' ) {
    message = 'Please include a valid email address.';
  }

  if ( type === 'checkgroup' ) {
    message = 'Please select at least one of the ' + label + ' options.';
  }

  $( elem ).trigger( 'cf_notifier:notify', {
    message: message
  } );
  $( field.elem ).addClass( 'error' );
}

function _clearNotification( elem ) {
  var $elem = $( elem );
  $elem.trigger( 'cf_notifier:clear' );
  $elem.find('.error').removeClass( 'error' );
}

module.exports = { init: init };

```

Here and throughout. See https://github.com/cfpb/cfgov-refresh/blob/flapjack/src/static/js/modules/util/web-storage-proxy.js for reference.